src/actions/events.js

```diff
@@ -228,6 +231,20 @@ events.on(POST_BUILD, function(event) {
     });
 });
 
+
+events.on(POST_REPO_SECRET, function(event) {
+  const {owner, name, secret} = event.data;
+  Request.post(`/api/repos/${owner}/${name}/secrets`)
+    .set('X-CSRF-TOKEN', token)
+    .send(secret)
+    .end((err) => {
+      if (err != null) {
+        console.error(err);
+      }
+      events.emit(GET_REPO_SECRETS, { owner, name });
```

Comment:
We should not need to emit events here. Updating the tree will cause the user interface to refresh.